### PR TITLE
SAK-43196 msgcntr: Deprecated utils.FormattedText, Validator => api.F…

### PR DESCRIPTION
**Pull request by axxter99**
…ormattedText

### CODE REVIEW

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java [hunk 1]
```diff
@@ -4159,7 +4161,7 @@ public String getAuthorString()
     
     public String getPlacementId() 
     {
-       return Validator.escapeJavascript("Main" + toolManager.getCurrentPlacement().getId());
+       return formattedText.escapeJavascript("Main" + toolManager.getCurrentPlacement().getId());
```
Comment (adrianfish) [inline msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java]: I'm not sure this is right. Validator.escapeJavascript is marked as deprecated, but the doc says to use StringEscapeUtils (Apache) instead. FormattedText doesn't use StringEscapeUtils. I'd either leave Validator here, or move FormattedText to use Apache (with the testing that'll need).

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/RankManagerImpl.java [hunk 2]
```diff
@@ -362,7 +364,7 @@ private String resourceUrlEscaping(String url) {
         String leftOfAttachment = url.substring(0, attIndex);
         String rightOfAttachment = url.substring(attIndex);
 
-        String finalUrl = leftOfAttachment.concat(Validator.escapeUrl(rightOfAttachment));
+        String finalUrl = leftOfAttachment.concat(formattedText.escapeUrl(rightOfAttachment));
```
Comment (adrianfish) [inline msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/RankManagerImpl.java]: Again, I'd leave this as Validator. FormattedText should be using URLEncoder, as per the doc in Validator.java.

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/entity/MessageForumsEntityContentProducer.java [hunk 3]
```diff
@@ -54,6 +56,8 @@
 	// runtime dependency
 	private List removeEvents = null;
 	
+	@Setter private FormattedText formattedText;
```
Comment (adrianfish) [inline msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/entity/MessageForumsEntityContentProducer.java]: Why not go the whole hog and remove the setters for those other private vars in there? Just move @Setter to the class definition and strip those setters out. You may as well.

File: msgcntr/messageforums-component-impl/src/webapp/WEB-INF/components.xml [hunk 4]
```diff
@@ -598,27 +599,27 @@
...
+                <property name="eventTrackingService">
+                    <ref bean="org.sakaiproject.event.api.EventTrackingService"/>
+                </property>
+                <property name="discussionForumManager">
+                  <ref bean="org.sakaiproject.api.app.messageforums.ui.DiscussionForumManager"/>
+                </property>
+                <property name="toolManager">
+                  <ref bean="org.sakaiproject.tool.api.ToolManager"/>
+                </property>
+            </bean>
```
Comment (adrianfish) [inline msgcntr/messageforums-component-impl/src/webapp/WEB-INF/components.xml]: I'd just compress all these onto one line, personally. No biggy though.

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ForumsEmailService.java [hunk 5]
```diff
@@ -56,8 +54,10 @@
 import org.sakaiproject.time.api.UserTimeService;
 import org.sakaiproject.tool.api.ToolManager;
 import org.sakaiproject.tool.messageforums.ui.DiscussionMessageBean;
-import org.sakaiproject.util.FormattedText;
 import org.sakaiproject.util.ResourceLoader;
+import org.sakaiproject.util.api.FormattedText;
+
+import lombok.extern.slf4j.Slf4j;
```
[the remark this answers is not in the document]
Comment (axxter99) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ForumsEmailService.java]: @Slf4j:  ForumsEmailService log. : 18 matches

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ForumsEmailService.java [hunk 6]
```diff
@@ -30,17 +30,15 @@
...
 import org.sakaiproject.api.app.messageforums.AnonymousManager;
 import org.sakaiproject.api.app.messageforums.Attachment;
 import org.sakaiproject.api.app.messageforums.BaseForum;
 import org.sakaiproject.api.app.messageforums.Message;
 import org.sakaiproject.api.app.messageforums.Topic;
 import org.sakaiproject.component.cover.ComponentManager;
 import org.sakaiproject.component.cover.ServerConfigurationService;
-import org.sakaiproject.content.api.ContentResource;
 import org.sakaiproject.content.api.ContentHostingService;
+import org.sakaiproject.content.api.ContentResource;
```
[the remark this answers is not in the document]
Comment (axxter99) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ForumsEmailService.java]: OK

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/RankManagerImpl.java (hunk 2 above)
Comment (axxter99) [reply msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/RankManagerImpl.java]: OK

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java (hunk 1 above)
Comment (axxter99) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java]: copmmons-text: StringEscapeUtils.escapeEcmaScript(String)?

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ForumsEmailService.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (axxter99) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ForumsEmailService.java]: OK

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/entity/MessageForumsEntityContentProducer.java (hunk 3 above)
Comment (axxter99) [reply msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/entity/MessageForumsEntityContentProducer.java]: OK

File: msgcntr/messageforums-component-impl/src/webapp/WEB-INF/components.xml (hunk 4 above)
Comment (axxter99) [reply msgcntr/messageforums-component-impl/src/webapp/WEB-INF/components.xml]: OK

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 11
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=4969278&bid=16417347)

**Pull request merged**